anvio/bottleroutes.py

```diff
@@ -1038,6 +1038,11 @@ def gen_summary(self, collection_name):
         # common params. we will set pan/profile specific params a bit later:
         summarizer_args.collection_name = collection_name
         summarizer_args.taxonomic_level = self.interactive.taxonomic_level
+        init_gene_coverages = request.forms.get('init_gene_coverages')
+        
+        if init_gene_coverages:
+            summarizer_args.init_gene_coverages = True
+            self.interactive.mode == 'full'
```

Comment:
Best practice would suggest that this is *not* a good place to do it. You are changing the `self.interactive.mode` to `full` if `init_gene_coverages` is present in POST variables.

There is an `elif self.interactive.mode == 'full'` below, and the check for `init_gene_coverages` should be handled under there. I.e.,

``` python
        (...)
        elif self.interactive.mode == 'full':
            summarizer_args.profile_db = self.interactive.profile_db_path
            summarizer_args.contigs_db = self.interactive.contigs_db_path

            if init_gene_coverages:
                summarizer_args.init_gene_coverages = True

            summarizer_args.output_dir = os.path.join(os.path.dirname(summarizer_args.profile_db), 'SUMMARY_%s' % collection_name)
        else:
            (...)
```